migration-guide.md

```diff
@@ -0,0 +1,119 @@
+# Migration guide
+
+This migration guide details how to migrate from from earlier Dart platform APIs
+in `package:platform` version 3.x to, and in `dart:io`, to `package:platform`
+version 4.x.
+
+## From `package:platform` version 3.x
+
+### General platform APIs
+
+The general APIs in `package:platform` v3.x, that determine what the host
+platform is, rely on instantiating an instance of the `LocalPlatform` class. In
+v4.0 a new convenience `.current` getter can be used, which returns the current
+host platform.
+
+Migrate from:
+```dart
+import 'package:platform/platform.dart'; // version 3.x
+
+bool onAndroid = LocalPlatform().isAndroid;
+```
+
+To:
+```dart
+import 'package:platform/platform.dart'; // version 4.x
+
+bool onAndroid = Platform.current.isAndroid;
```

Comment:
Should actually be
```dart
import 'package:platform/host.dart';

bool onAndroid = HostPlatform.current.isAndroid;
```

The `platform.dart` library is not cross-platform capable, it's there for backwards compatibility with v3, where `platform.dart` allowed access to native-only features.

Reply:
I would like it if we could do the full migration to `HostPlatform`, but whether that is realistic I think depends on whether `dart fix` can automate the migration. I'm going to share this page with the analyzer team to get their input.